### PR TITLE
Gpudev bjorn

### PR DESCRIPTION
**Pull request by liruipeng**
GPU Ext/Ext+i interpolations

### CODE REVIEW

**Review by ulrikeyang — APPROVED** [review]
Looks good to me, although I trust that the CUDA changes are correct.

File: src/parcsr_ls/par_lr_interp.c [hunk 1]
```diff
@@ -591,9 +576,7 @@ hypre_BoomerAMGBuildStdInterp(hypre_ParCSRMatrix  *A,
             {
                indx = ihat[i1];
                if (indx > -1)
-               {
                   ahat[indx] += A_diag_data[jj];
-               }
```
Comment (rfalgout) [inline src/parcsr_ls/par_lr_interp.c]: Why are all of the open/close squirly braces being removed?  Didn't you just go through a big effort to put them in?

**Review by rfalgout — APPROVED** [review]
Looks good, though I didn't understand why the '{' and '}' delimiters were removed in many places.  In general, the convention is to always put these in to avoid accidental logic issues.  We can wait to fix this later, though.

**Comment by liruipeng** [comment]
> Looks good, though I didn't understand why the '{' and '}' delimiters were removed in many places. In general, the convention is to always put these in to avoid accidental logic issues. We can wait to fix this later, though.

I really don't know how it could happen. I had to manually add them back...

**Review by bjorn2 — APPROVED** [review]
Well, lot of changes.  Looks ok, assume you have tested that the code still gives correct results.

**Pull request merged**